flink-dist/src/main/assemblies/opt.xml

```diff
@@ -75,6 +75,28 @@
 			<fileMode>0644</fileMode>
 		</file>
 
+		<!-- External Resource -->
+		<file>
+			<source>../flink-external-resources/flink-external-resource-gpu/target/flink-external-resource-gpu-${project.version}.jar</source>
+			<outputDirectory>opt/external-resource-gpu/</outputDirectory>
+			<destName>flink-external-resource-gpu-${project.version}.jar</destName>
+			<fileMode>0644</fileMode>
+		</file>
+
+		<file>
+			<source>../flink-external-resources/flink-external-resource-gpu/src/main/resources/gpu-discovery-common.sh</source>
+			<outputDirectory>opt/external-resource-gpu/</outputDirectory>
+			<destName>gpu-discovery-common.sh</destName>
+			<fileMode>0755</fileMode>
+		</file>
+
+		<file>
+			<source>../flink-external-resources/flink-external-resource-gpu/src/main/resources/nvidia-gpu-discovery.sh</source>
+			<outputDirectory>opt/external-resource-gpu/</outputDirectory>
+			<destName>nvidia-gpu-discovery.sh</destName>
+			<fileMode>0755</fileMode>
+		</file>
```

Comment:
I'm wondering whether we should add the external gpu support right away to the Flink binary distribution or whether it would also be ok to download it from the Flink web page? How large are the binaries we are adding here?

Reply:
Do you worried about the total size of flink-dist? Since this feature is only 22 kb (while most of the metric reporter plugin are over 100 kb), I think it would not affect the time of downloading flink-dist. I also notice that there is a discussion about offering slim jar and fat jar. If I understand it correctly, it would be good to just include it to the fat jar.
Regarding to the usability, I think it would be easier to move the `external-resource-gpu` from 'opt' to 'plugin' than searching and downloading it from website. WDYT?

Reply:
I think we can do it at the moment. We might revisit this topic once we properly define what a slim and fat release is.